src/Features/Core/Portable/MetadataAsSource/DecompilationMetadataAsSourceFileProvider.cs

```diff
@@ -125,18 +126,14 @@ public DecompilationMetadataAsSourceFileProvider(IImplementationAssemblyLookupSe
                 // Create the directory. It's possible a parallel deletion is happening in another process, so we may have
                 // to retry this a few times.
                 var directoryToCreate = Path.GetDirectoryName(fileInfo.TemporaryFilePath)!;
+                var loopCount = 0;
                 while (!Directory.Exists(directoryToCreate))
                 {
-                    try
-                    {
-                        Directory.CreateDirectory(directoryToCreate);
-                    }
-                    catch (DirectoryNotFoundException)
-                    {
-                    }
-                    catch (UnauthorizedAccessException)
-                    {
-                    }
+                    // Protect against infinite loops.
+                    if (loopCount++ > 10)
+                        return null;
```

Comment:
>null [](http://example.com/codeflow?start=31&length=4)

Should we log the error somewhere?

Reply:
Yes probably. I thought this would just end up at the standard handling for non-navigable symbols, but all of that code seems to have changed since I last looked for it, and I think its actually ignoring the nullable result from this method, which is not good. Gonna close this and try again some other time.